_tutorials/basic-csharp-mod.md

```diff
@@ -6,14 +6,31 @@ The following guide will walk you through step-by-step on how to create a basic
 
 ## Preparation
 
+### Code environment
+
+#### Important!
+There is a fantastic set of four videos on youtube that run you through the mod setup process. While this tutorial serves as a slightly more in-depth text version, you should definitely watch these videos:
+https://www.youtube.com/playlist?list=PL-eEdYu_CymFEGjndmZRmM-gSg8AK66L_
+
+Or, read on for the more in-depth version.
+
+We will be using Visual Studio 2019 and DotNet Framework 4.7.2. If you already know how to set up a DotNet project and add references from .dll files, feel free to skip this part of the tutorial and jump ahead to "Setting up your Module".
+
+Firstly, download Visual Studio Community 2019. Go to this link and click "Free Download" under the "Community" option: https://visualstudio.microsoft.com/downloads/. This is one of those fantastic microsoft things that installs a downloader that downloads an installer to download the files you need to install a downloader that downloads an installer for the downloader that you need to install the downloader that... you get the idea.
```

Comment:
How about mention JetBrains Rider?

Reply:
What's that?

Reply:
https://www.jetbrains.com/rider/
It's good IDE for C#, not so heavy and terrible like  VS.

Reply:
I used a IDE Rider and that is why easy to build a project